### PR TITLE
ONNX Export Scripted Interpolate Op

### PR DESCRIPTION
**Pull request by lara-hdr**
We currently support exporting traced interpolate ops to ONNX.

Scripting interpolate op invokes aten::__interpolate in the Torch IR (instead of aten::upsample_[mode][dim]d), which we do not support yet.
This PR implements the ONNX symbolic for __interpolate() to support exporting interpolate in scripting scenarios. 

Related open issue: https://github.com/pytorch/pytorch/issues/25807

### CODE REVIEW

File: torch/onnx/symbolic_opset8.py [hunk 1]
```diff
@@ -74,6 +74,22 @@ def symbolic_fn(g, input, output_size, align_corners=None):
 upsample_trilinear3d = _interpolate('upsample_trilinear3d', 5, "linear")
 
 
+def __interpolate(g, input, size, scale_factor, mode , align_corners):
+    align_corners = sym_help._maybe_get_const(align_corners, 'b')
+    if not align_corners.node().mustBeNone() and align_corners:
+        return _unimplemented("interpolate", "align_corners == True")
+
+    if not scale_factor.node().mustBeNone() and sym_help._is_value(scale_factor):
```
Comment (houseroad) [inline torch/onnx/symbolic_opset8.py]: Can we extract this to some helper function? So we don't need to repeat this pattern manually.

Comment (houseroad) [reply torch/onnx/symbolic_opset8.py]: Feel free to do it in a dedicated refactor PR.

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -247,19 +249,65 @@ def _interpolate_size_to_scales(g, input, output_size, dim):
     output_size = _maybe_get_const(output_size, 'is')
     if _is_value(output_size):
         offset = 2
-        offsets = g.op("Constant", value_t=torch.ones(offset))
+        offsets = g.op("Constant", value_t=torch.ones(offset, dtype=torch.float32))
         dividend = g.op("Cast", output_size, to_i=cast_pytorch_to_onnx["Float"])
-        divisor = _slice_helper(g, g.op("Shape", input), axes=[0], ends=[dim], starts=[offset])
+        divisor = _slice_helper(g, g.op("Shape", input), axes=[0], ends=[9223372036854775807], starts=[offset])
```
Comment (houseroad) [inline torch/onnx/symbolic_helper.py]: Can we use some constant like double.max to replace this number? it's so error prone.

File: torch/onnx/symbolic_helper.py [hunk 3]
```diff
@@ -247,19 +249,65 @@ def _interpolate_size_to_scales(g, input, output_size, dim):
...
         scales = g.op("Concat", offsets, scale_dims, axis_i=0)
     else:
         scales_constant = [1. if i < 2 else
                            float(output_size[-(dim - i)]) / float(input.type().sizes()[-(dim - i)])
                            for i in range(0, dim)]
-        scales = g.op("Constant", value_t=torch.tensor(scales_constant))
+        scales = g.op("Constant", value_t=torch.tensor(scales_constant, dtype=torch.float32))
     return scales
 
+def _interpolate_get_scales(g, scale_factor, dim):
```
Comment (houseroad) [inline torch/onnx/symbolic_helper.py]: make _interpolate_get_scales and _interpolate_get_scales_and_mode_from_args consistent?

_interpolate_get_scales + _interpolate_get_scales_and_mode or 
_interpolate_get_scales_from_args + _interpolate_get_scales_and_mode_from_args

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -485,7 +520,9 @@ def _interpolate_tests(self, is_upsample):
                         mode_i = "trilinear"
                 self._interpolate(xi, mode_i, True, is_upsample)
                 if self.opset_version >= 9:  # throws unimplemented
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: update the comments?

**Review by houseroad — APPROVED** [review]
Looks good.
